base/src/instantiate.rs

```diff
@@ -8,6 +8,11 @@ use types;
 use types::{AliasData, BuiltinType, Type, Generic, TcType, TypeEnv, merge};
 use symbol::Symbol;
 
+pub enum Error<I> {
+    SelfRecursive(I),
+    UndefinedType(I),
+}
```

Comment:
Thats a bit sloppy of me, only needed a way to distinguish the two error cases from each other as the only place I use it is to convert it into another error. Should be quick to add though.